src/main/java/org/terasology/simpleFarming/components/TreeGrowthStage.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.simpleFarming.components;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.terasology.math.geom.Vector3i;
+import org.terasology.reflection.MappedContainer;
+
+/**
+ * Represents a stage of growth for a tree.
+ * <p>
+ * Contains the positions of leaves relative to the root block and the number of logs this growth stage should contain.
+ * 
+ * @see SaplingDefinitionComponent
+ */
+@MappedContainer
+public class TreeGrowthStage {
+    /** Number of logs this stage should have. */
+	public int height;
+	
+	/** Locations of each leaf in this stage, relative to the root (the lowest log block). */
+	public Set<Vector3i> leaves = new HashSet<>();
+    
+    /**
+     * Default leaf configuration to use, 0 if custom defined.
+     * There are currently 3 default leaf configurations.
+     */
+    public int defLeaves;
+
+    /** Minimum time before this growth stage, in milliseconds. */
+    public int minTime;
+
+    /** Maximum time before this growth stage, in milliseconds. */
+    public int maxTime;
+
+    /** Default constructor required for persistence. */
+    public TreeGrowthStage() {
+    }
+
+    /** Construct a copy of the given {@code GrowthStage}. */
+    public TreeGrowthStage(TreeGrowthStage clone) {
+    	this.height = clone.height;
+    	this.leaves = clone.leaves;
+        this.maxTime = clone.maxTime;
+        this.minTime = clone.minTime;
+    }
+    
+    /**
+     * Gets the leaf locations for this growth stage.
+     * If using a default configuration, the leaves will automatically be adjusted to the {@link #height}
+     * 
+     * @return A Set of all of the locations where a leaf should be grown.
+     */
+    public Set<Vector3i> getLeaves() {
+    	if(defLeaves == 0) {
+    		return leaves;
+    	}
+    	
+    	Vector3i location = new Vector3i(0, height-1, 0);
+    	Set<Vector3i> defLeafSet = new HashSet<>();
+    	switch(defLeaves) {
+    		case 3:
```

Comment:
I'm not going to request this for GCI but I'm not a huge fan of this type of default specification.

What I'd propose is to actually move the leaf structure into a separate component. In that component I'd even be fine with providing only one default configuration hard coded into the `leaves` field of that component.
That way a dev could make a few different tree designs, and place the component onto a template prefab to be inherited from. Similar to the Base Bush.

If there is no such leaf component on the tree then use an appropriate default. Either by adding a blank leaf component (with the default in it) or by hard coding a default into the Tree System like this method.

If you felt like it you could also then split off the trunk design into an _optional_ separate component as well but this would be a stylistic choice I'd leave up to you. It'd be perfectly reasonable for simple farming to be limited to altering the leaves and not the trunk as well.

Reply:
I made it so that each growth stage contains the name of a prefab with a LeafStructureComponent. There are 3 prefabs included for default small, medium, and large leaves.